### PR TITLE
C++ error message cleanup

### PR DESCRIPTION
**Pull request by speth**
I had noticed a few instances where calls to `CanteraError` referenced the wrong method name (usually due to copy-paste), and decided to do a full pass over all of the places where `CanteraError` and related exceptions are thrown to fix these errors, updating them to always include both the class name and method name, as well as normalizing a few variations on the formatting.

This also updates the documentation for the `CanteraError` class to specify how the method name should be written.

### CODE REVIEW

**Comment by bryanwweber** [comment]
A few of these changes (at least some of the ones for Debye-Huckel) I've already made in #693. It would be easier for me if you could drop those changes from here 😄

**Comment by bryanwweber** [comment]
https://github.com/Cantera/cantera/pull/693/files?file-filters%5B%5D=.cpp#diff-cf7ca91b65f695af8cde3564c35d1ad6R122-R131 and the `IdealMolalSoln` file right below that.

File: src/kinetics/Kinetics.cpp [hunk 1]
```diff
@@ -326,7 +328,7 @@ thermo_t& Kinetics::speciesPhase(const std::string& nm)
             return thermo(n);
         }
     }
-    throw CanteraError("speciesPhase", "unknown species "+nm);
+    throw CanteraError("Kinetics::speciesPhase", "unknown species "+nm);
```
Comment (ischoegl) [inline src/kinetics/Kinetics.cpp]:
```
throw CanteraError("Kinetics::speciesPhase", "unknown species {}", nm);
```
Use python-style formatting?

File: src/base/units.h [hunk 2]
```diff
@@ -121,7 +121,7 @@ class Unit
             // tok is not one of the entries in map m_u, then
             // m_u[tok] returns 0.0. Check for this.
             if (fctr == 0) {
-                throw CanteraError("toSI","unknown unit: "+tsub);
+                throw CanteraError("Unit::toSI","unknown unit: "+tsub);
```
Comment (ischoegl) [inline src/base/units.h]: Space after comma / Python style formatting?

**Review by ischoegl — COMMENTED** [review]
Noticed some missing spaces, and python-style formatting *could* be more consistent (although it's debatable of whether this should be done for code that's not really being touched).

Other than that, some error messages could be more meaningful, especially some that are triggered during setup of thermo/kinetics objects. E.g. it may not be clear what thermo phase triggers a message like `"Species '{}' not found"`, i.e. it would be great if the context could be provided, e.g.
```C++
throw CanteraError("obj::method", "Phase '{}' does not contain species '{}'", phaseName, speciesName);
```

**Comment by speth** [comment]
> A few of these changes (at least some of the ones for Debye-Huckel) I've already made in #693.

This actually rebased cleanly onto `master` after your PR was merged, so I guess we made exactly the same edits. :tada: 

> Noticed some missing spaces, and python-style formatting could be more consistent

Thanks for the reminder -- I fixed this for all of the lines that were affected by this PR.

> Other than that, some error messages could be more meaningful, especially some that are triggered during setup of thermo/kinetics objects.

Certainly not a bad idea, I think this is well outside the scope of what I'm trying to clean up here.

File: include/cantera/base/ctexceptions.h [hunk 3]
```diff
@@ -62,8 +62,12 @@ class CanteraError : public std::exception
 public:
     //! Normal Constructor for the CanteraError base class
     /*!
-     * @param procedure String name for the function within which the error was
-     *             generated.
+     * @param procedure Name of the function within which the error was
+     *             generated. For member functions, this should be written as
+     *             `ClassName::functionName`. For constructors, this should be
```
Comment (bryanwweber) [inline include/cantera/base/ctexceptions.h:67]: Does the single backtick do something special in doxygen?

File: src/thermo/PDSS_SSVol.cpp [hunk 4]
```diff
@@ -28,23 +28,25 @@ void PDSS_SSVol::setParametersFromXML(const XML_Node& speciesNode)
...
+                           "no 'standardState' Node for species '{}'",
+                           speciesNode.name());
     }
     std::string model = ss->attrib("model");
     vector_fp coeffs;
     getFloatArray(*ss, coeffs, true, "toSI", "volumeTemperaturePolynomial");
     if (coeffs.size() != 4) {
         throw CanteraError("PDSS_SSVol::setParametersFromXML",
-                           " Didn't get 4 density polynomial numbers for species " + speciesNode.name());
+                           " Didn't get 4 density polynomial numbers for species '{}'",
```
Comment (bryanwweber) [inline src/thermo/PDSS_SSVol.cpp]: Leading space on this line

**Review by bryanwweber — APPROVED** [review]
Looks good to me, just some very minor comments

File: include/cantera/base/ctexceptions.h (hunk 3 above)
Comment (speth) [reply include/cantera/base/ctexceptions.h:67]: It's the way to format something as code using a fixed-width font, like in Markdown.

Comment (bryanwweber) [reply include/cantera/base/ctexceptions.h:67]: I really wish reST used the same syntax. Ah well...

**Pull request merged**